File: sdks/python/windjammer_sdk/audio.py
"""Audio components."""

from .math import Vec3


class AudioSource:
    """Audio source component."""
    
    def __init__(self, audio_file: str = "", volume: float = 1.0, looping: bool = False):
        self.audio_file = audio_file
        self.volume = volume
        self.looping = looping
        self.position = Vec3.zero()
    
    def play(self) -> None:
        """Play the audio."""
        print(f"[Audio] Playing: {self.audio_file}")
    
    def stop(self) -> None:
        """Stop the audio."""
        print(f"[Audio] Stopping: {self.audio_file}")
    
    def __repr__(self) -> str:
        return f"AudioSource(file='{self.audio_file}', volume={self.volume})"


class AudioListener:
    """Audio listener component."""
    
    def __init__(self, position: Vec3 = None):
        self.position = position or Vec3.zero()
    
    def __repr__(self) -> str:
        return f"AudioListener(pos={self.position})"

File: sdks/javascript/src/audio.ts
/**
 * Audio components.
 */

import { Vec3 } from './math';

/**
 * Options for creating an audio source.
 */
export interface AudioSourceOptions {
  /** Audio file path */
  audioFile?: string;
  /** Volume level */
  volume?: number;
  /** Whether to loop */
  looping?: boolean;
}

/**
 * Audio source component.
 */
export class AudioSource {
  /** Audio file path */
  audioFile: string;
  
  /** Volume level */
  volume: number;
  
  /** Whether to loop */
  looping: boolean;
  
  /** Position in 3D space */
  position: Vec3;

  /**
   * Create a new audio source.
   * 
   * @param options - Audio source options
   */
  constructor(options: AudioSourceOptions = {}) {
    this.audioFile = options.audioFile || '';
    this.volume = options.volume ?? 1.0;
    this.looping = options.looping ?? false;
    this.position = Vec3.zero();
  }

  /**
   * Play the audio.
   */
  play(): void {
    console.log(`[Audio] Playing: ${this.audioFile}`);
  }

  /**
   * Stop the audio.
   */
  stop(): void {
    console.log(`[Audio] Stopping: ${this.audioFile}`);
  }

  toString(): string {
    return `AudioSource(file='${this.audioFile}', volume=${this.volume})`;
  }
}

/**
 * Audio listener component.
 */
export class AudioListener {
  /** Position in 3D space */
  position: Vec3;

  /**
   * Create a new audio listener.
   * 
   * @param position - Listener position
   */
  constructor(position: Vec3 = Vec3.zero()) {
    this.position = position;
  }

  toString(): string {
    return `AudioListener(pos=${this.position})`;
  }
}

